python/pyxet/pyxet/url_parsing.py

```diff
@@ -22,162 +24,229 @@ def parse_url(url, force_domain='xethub.com', partial_remote=False):
     If partial_remote==True, allows [repo] to be optional. i.e. it will
     parse /user or xet://user
     """
+
     url = url.lstrip('/')
     parse = urlparse(url)
     if parse.scheme == '':
         parse = parse._replace(scheme='xet')
 
-    # support force_domain with a scheme (http/https)
-    domain_split = force_domain.split('://')
+    # support default_domain with a scheme (http/https)
+    domain_split = default_domain.split('://')
     scheme = 'https'
     if len(domain_split) == 2:
         scheme = domain_split[0]
-        force_domain = domain_split[1]
+        default_domain = domain_split[1]
 
     if parse.scheme != 'xet':
         raise ValueError('Invalid protocol')
 
     # Handle the case where we are xet://user/repo. In which case the domain
     # parsed is not xethub.com and domain="user".
     # we rewrite the parse the handle this case early.
-    if parse.netloc != force_domain:
-        if parse.netloc == 'xethub.com':
-            parse = parse._replace(netloc=force_domain)
-        else:
-            # this is of the for xet://user/repo/...
-            # join user back with path
-            newpath = f"/{parse.netloc}{parse.path}"
-            # replace the netloc
-            true_netloc = force_domain
-            parse = parse._replace(netloc=true_netloc, path=newpath)
+    if "@" not in parse.netloc:
+        if not recursive_skip_checks:
+            global has_warned_user_on_url_format
+            
+            if not has_warned_user_on_url_format:
+                sys.stderr.write("Warning:  The use of the xet:// prefix without an endpoint is deprecated and will be disabled in the future.\n"
+                                f"          Please switch URLs to use the format xet://<user>@<endpoint>/<repo>/<branch>/<path>.\n"
+                                f"          Endpoint now defaulting to {default_domain}.\n\n")
+                has_warned_user_on_url_format = True
+
+    else:
+        # Hack until we can clean up this parsing logic
+        user_at_domain = parse.netloc.split("@")
+        if len(user_at_domain) == 2:
+            user, domain = user_at_domain
+            # In this case, swap out the url to the old form and tell it to reparse it.  
+            # This is easier than trying to figure out how to rework the url parsing tool logic
+            new_url = url.replace(parse.netloc, f"{domain}/{user}")
+            return parse_url(new_url, default_domain=domain, partial_remote=partial_remote, recursive_skip_checks=True)
+        else: 
+            raise ValueError(f"Cannot parse user and endpoint from {parse.netloc}")
+
+    if not parse.netloc.endswith(".com"):  # Cheap way now to see if it's a website or not; we won't hit this with the old format.
```

Comment:
What if we have a `xethub.uw.edu` deployment in the future, hmmm, just a guess

Reply:
This is only on the non-@ path with will be deprecated, so I'm not worried about future stuff.  This is just to see if the start is `xet://domain/user/...` or `xet://user/`  You can break this by having a user that ends in .com, but I'm also not worried about that now...